### DISCUSSION TITLE
(Done) July 9th: deps.fnl

### DISCUSSION THREAD

**Discussion by frenchy64** [madclj/madclj.com#16] [Announcements]
👍 to RSVP

https://madclj.com/

Wednesday, July 9th, 2025
6:30 PM to 9:00 PM CT

[StartingBlock Madison](https://www.google.com/maps/search/?api=1&query=43.081245%2C%20-89.374054)
821 E Washington Ave 2nd floor · Madison, WI

[Fennel](https://fennel-lang.org/) is the closest thing to a [Clojure for Lua](https://fennel-lang.org/from-clojure). Recently, the [deps.fnl](https://gitlab.com/andreyorst/deps.fnl) dependency manager was released, inspired by the Clojure CLI. It provides git deps and Luarocks integration (Lua packages).

Ideas for discussion:
- deps.fnl vs deps.edn
  - special consideration for Luarocks transitive deps vs Maven transitive deps
    - version ranges thwart reproducibility
  - ability to provide e.g., `:paths` for deps
    - depend on Lua libs via git that don't know about deps.fnl
- Q: Many Clojure variants can use deps.edn (cljs, Clojure Dart, jank). Clojure CLR is an exception. Why do Clojure CLR and Fennel need their own tooling?
- deps.fnl blog posts
  - https://andreyor.st/posts/2025-01-10-depsfnl-a-new-dependency-manager-for-fennel-projects/
  - https://andreyor.st/posts/2025-02-16-depsfnl-022-released/
- Fennel vs CLJS vs Cherry
  - Fennel/Cherry are direct translations to Lua/JS
  - CLJS has advanced compilation options, but is tethered to Google Closure's fate
- [Fennel Tutorial](https://fennel-lang.org/tutorial)
- [Options for deploying Fennel code](https://fennel-lang.org/setup#embedding-fennel)
  - similar options to cljs
    - uberscript
    - embedded compiler/repl
- [fennel-cljlib](https://gitlab.com/andreyorst/fennel-cljlib)
  - Clojure-like library for Fennel
- [8fl](https://git.sr.ht/~nasser/8fl-renoise)
  - Fennel live-coding for [Renoise](https://www.renoise.com/)
  - [Video: Fennel Conf 2022](https://conf.fennel-lang.org/2022#video7)
- [ambrosebs-reaper-scripts](https://github.com/frenchy64/ambrosebs-reaper-scripts/tree/main/fnl)
  - combines bb + clj + fnl => Lua scripts for [Reaper](https://www.reaper.fm/)
- [renoise-fennel-starter-kit](https://git.sr.ht/~ambrosebs/renoise-fennel-starter-kit)
  - example base for scripting [Renoise](https://www.renoise.com/) with Fennel

**Comment by frenchy64** [comment]
@madclj/meetup this week we'll be looking into Fennel's new dependency tool deps.fnl, inspired by Clojure CLI. We can use this as a bouncing off point for lots of different topics, like learning more about Clojure CLI or Lua, or comparing Fennel with CLJS. I listed some ideas in the thread.

I was also involved in the early stages of deps.fnl, so I could also give some perspective about the tool's evolution.

**Reply by brunchboy** [reply]
It's the part of the summer where Madison Chamber Orchestra plays concerts on the square on Wednesday evenings, and we're responsible for setting out a blanket for a big group of friends since we live right on the square, so I can't make it, though it sounds interesting.

**Reply by sivakusayan** [reply]
I'll be there in spirit :)

**Comment by ericnormand** [comment]
I'll be there alone (no kids).

**Comment by frenchy64** [comment]
I'm working on combining the Renoise <=> Fennel bridge with my Fennel <=> Clojure bridge, I might have something to show scripting Renoise from Clojure.

**Comment by frenchy64** [comment]
Ordered a couple of pizzas, bussing over. I'll get some snacks at Festival if there's time.
